### PR TITLE
[16.0][IMP] purchase_security: Add groups to purchase orders and new group permission feature + Restrict partners according to purchase group

### PR DESCRIPTION
**Pull request by victoralmau**
FWP from 15.0: https://github.com/OCA/purchase-workflow/pull/1921 + https://github.com/OCA/purchase-workflow/pull/2153

Changes done:
- [x] Add groups to purchase orders and new group permission feature
- [x] Restrict partners according to purchase group
- [x] Adapt to 16.0

Please @pedrobaeza and @chienandalu can you review it?

@Tecnativa TT47237

### CODE REVIEW

**Review by pedrobaeza — APPROVED** [review]
You can squash my 2 commits into one.

File: purchase_security/models/purchase_order.py [hunk 1]
```diff
@@ -10,9 +13,41 @@ class PurchaseOrder(models.Model):
...
+        assigned, and if no one, the first purchase team.
+        """
+        first_team = self.env["purchase.team"].search([], limit=1)
+        for record in self:
+            record.team_id = record.user_id.purchase_team_ids[:1] or first_team
+
+    def onchange_partner_id(self):
+        res = super().onchange_partner_id()
+        if self.partner_id:
+            partner = self.partner_id or self.partner_id.commercial_partner_id
```
Comment (yostashiro) [inline purchase_security/models/purchase_order.py]: I think it was intended this way?
```suggestion
            partner = self.partner_id.commercial_partner_id
```

File: purchase_security/views/res_partner_views.xml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    <record id="view_partner_property_form" model="ir.ui.view">
+        <field name="name">res.partner.property.form.inherit</field>
+        <field name="model">res.partner</field>
+        <field name="inherit_id" ref="account.view_partner_property_form" />
+        <field name="arch" type="xml">
+            <!-- Use this field to place the field first in Purchase group. !-->
+            <field name="property_supplier_payment_term_id" position="before">
+                <field
+                    name="purchase_user_id"
+                    groups="purchase_security.group_purchase_own_orders"
```
Comment (yostashiro) [inline purchase_security/views/res_partner_views.xml:49]:
```suggestion
                    groups="purchase_security.group_purchase_own_orders"
                    widget="many2one_avatar_user"
```

File: purchase_security/views/purchase_team_views.xml [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        </field>
+    </record>
+    <record id="purchase_team_tree" model="ir.ui.view">
+        <field name="name">purchase.team.form</field>
+        <field name="model">purchase.team</field>
+        <field name="arch" type="xml">
+            <tree editable="bottom">
+                <field name="sequence" widget="handle" />
+                <field name="name" />
+                <field name="user_ids" widget="many2many_tags" />
```
Comment (yostashiro) [inline purchase_security/views/purchase_team_views.xml]:
```suggestion
                <field name="user_ids" widget="many2many_avatar_user" domain="[('share', '=', False)]" />
```

**Review by yostashiro — COMMENTED** [review]
Nice improvement. Would you consider updating the readme?

**Comment by victoralmau** [comment]
Changes done.

File: purchase_security/readme/DESCRIPTION.rst [hunk 4]
```diff
@@ -1,4 +1,6 @@
-This addon creates a new group called "Purchase (own orders)" in Purchase.
+This addon creates new groups in Purchase.
 
-For users in this group, the only Purchase Orders they can see are those where
-they are the representative, or all of them if they are managers.
+- User (own orders): If you are a follower of the partner or there is no user
```
Comment (yostashiro) [inline purchase_security/readme/DESCRIPTION.rst:6]: To make the explanation a bit clearer.

```suggestion
Visibility of purchase orders is restricted for users in these groups. You can only see the purchase order:

- User (own orders): If you are a follower of the partner or there is no user
```

**Review by yostashiro — APPROVED** [review]
Thanks for the update!

**Review by pilarvargas-tecnativa — APPROVED** [review]

**Pull request merged**